pkg/distributors/contracts/PoolTokenManipulator.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/EnumerableSet.sol";
+
+contract PoolTokenManipulator {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    IVault public immutable vault;
+
+    constructor(IVault _vault) {
+        vault = _vault;
+    }
+
+    mapping(bytes32 => EnumerableSet.AddressSet) private _poolTokenSets;
+    mapping(bytes32 => bool) private __poolTokenSetSaved;
+
+    function ensurePoolTokenSetSaved(bytes32 poolId) public {
```

Comment:
This could run into trouble when we have pools which don't have a fixed set of tokens, e.g. investment pools.

Reply:
I would like to avoid querying them every time.  This exiter contract is just an example of what could be done with the bpt when a user unstakes, so I'm not too worried about future proofing it.  Perhaps we add a method "refreshPoolTokenSet" that can be called manually when tokens are added?

Reply:
Ah yeah, this should be fine as the worst case is we deploy a new version and there's no lock in on this one.